### PR TITLE
[SYCL][ESIMD] Enable tests passing with new GPU driver

### PR DESCRIPTION
**Pull request by turinevgeny**

### CODE REVIEW

**Review by sarnex — APPROVED** [review]

**Comment by v-klochkov** [comment]
@turinevgeny - please resolve the conflicts

**Comment by turinevgeny** [comment]
> @turinevgeny - please resolve the conflicts

Done.

File: sycl/test-e2e/ESIMD/slm_init_no_inline.cpp [hunk 1]
```diff
@@ -27,6 +25,12 @@ int main() {
             << ", driver=" << D.get_info<info::device::driver_version>()
             << std::endl;
 
+  if (!isGPUDriverGE(Q, esimd_test::GPUDriverOS::LinuxAndWindows, "26516",
```
Comment (bader) [inline sycl/test-e2e/ESIMD/slm_init_no_inline.cpp:28]: @aelovikov-intel, do I recall correctly that we have a LIT feature to test against the driver version and report the test status correctly?

Comment (aelovikov-intel) [reply sycl/test-e2e/ESIMD/slm_init_no_inline.cpp:28]: Yes, since https://github.com/intel/llvm/pull/11427.

Comment (bader) [reply sycl/test-e2e/ESIMD/slm_init_no_inline.cpp:28]: @turinevgeny, I suggest using `REQUIRES-INTEL-DRIVER` instead this if-condition.

**Comment by turinevgeny** [comment]
@intel/llvm-gatekeepers, this PR is ready.

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**